tests/test_wsireader.py

```diff
@@ -68,60 +73,100 @@
 TEST_CASE_RGB_1 = [np.ones((3, 100, 100), dtype=np.uint8)]  # CHW
 
 
-class TestCuCIMReader(unittest.TestCase):
-    @skipUnless(has_cucim, "Requires CuCIM")
-    def setUp(self):
-        download_url(FILE_URL, FILE_PATH, "5a3cfd4fd725c50578ddb80b517b759f")
+def save_rgba_tiff(array: np.ndarray, filename: str, mode: str):
+    """
+    Save numpy array into a TIFF RGB/RGBA file
+
+    Args:
+        array: numpy ndarray with the shape of CxHxW and C==3 representing a RGB image
+        file_prefix: the filename to be used for the tiff file. '_RGB.tiff' or '_RGBA.tiff' will be appended to this filename.
+        mode: RGB or RGBA
+    """
+    if mode == "RGBA":
+        array = np.concatenate([array, 255 * np.ones_like(array[0])[np.newaxis]]).astype(np.uint8)
+
+    img_rgb = array.transpose(1, 2, 0)
+    imsave(filename, img_rgb, shape=img_rgb.shape, tile=(16, 16))
+
+    return filename
+
+
+@skipUnless(has_cucim or has_osl, "Requires cucim or openslide!")
+def setUpModule():  # noqa: N802
+    download_url(FILE_URL, FILE_PATH, "5a3cfd4fd725c50578ddb80b517b759f")
+
+
+class WSIReaderTests:
+    backend = None
 
     @parameterized.expand([TEST_CASE_0])
-    def test_read_whole_image(self, file_path, expected_shape):
-        reader = WSIReader("cuCIM")
+    def test_read_whole_image(self, file_path, level, expected_shape):
+        reader = WSIReader(self.backend, level=level)
         img_obj = reader.read(file_path)
         img = reader.get_data(img_obj)[0]
         self.assertTupleEqual(img.shape, expected_shape)
```

Comment:
looks like this class doesn't have the methods such as `self.assertTupleEqual`.

Reply:
This class are not being called directly so that would be fine but I'll change it.

Reply:
update that.

Reply:
@wyli, updated the tests. Is it good now?